client/components/data/query-post-likes/index.jsx

```diff
@@ -12,12 +12,23 @@ import { connect } from 'react-redux';
  * Internal dependencies
  */
 import { requestPostLikes } from 'state/posts/likes/actions';
+import getPostLikeLastUpdated from 'state/selectors/get-post-like-last-updated';
+import getPostLikes from 'state/selectors/get-post-likes';
 
 class QueryPostLikes extends Component {
 	static propTypes = {
 		siteId: PropTypes.number.isRequired,
 		postId: PropTypes.number.isRequired,
-		requestPostLikes: PropTypes.func,
+		needsLikers: PropTypes.bool,
+		hasPostLikes: PropTypes.bool,
+		maxAgeSeconds: PropTypes.number, // max age of likes data in milliseconds
+		lastUpdated: PropTypes.number, // timestamp of when the like data was last updated
+		requestPostLikes: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		maxAgeSeconds: 90,
```

Comment:
this seems offly short

Reply:
thoughts on a better timeframe? five minutes?

Reply:
i think ideally the API would support If-Modified-Since and we could get a 304 back if nothing had changed. Future work though.